frontend/javascripts/oxalis/model/bucket_data_handling/bucket_picker_strategies/oblique_bucket_picker.ts

```diff
@@ -3,103 +3,162 @@ import type { EnqueueFunction } from "oxalis/model/bucket_data_handling/layer_re
-  // sent to the GPU.
-  const scanLinesPoints = M4x4.transformVectorsAffine(
-    queryMatrix,
-    _.flatten(
-      _.range(steps + 1).map((idx) => [
-        // Cast lines at z=-10
-        [-enlargedHalfExtent, -enlargedHalfExtent + idx * stepSize, -10],
-        [enlargedHalfExtent, -enlargedHalfExtent + idx * stepSize, -10], // Cast lines at z=0
-        [-enlargedHalfExtent, -enlargedHalfExtent + idx * stepSize, 0],
-        [enlargedHalfExtent, -enlargedHalfExtent + idx * stepSize, 0], // Cast lines at z=10
-        [-enlargedHalfExtent, -enlargedHalfExtent + idx * stepSize, 10],
-        [enlargedHalfExtent, -enlargedHalfExtent + idx * stepSize, 10],
-      ]),
-    ),
-  );
-
-  let traversedBuckets = _.flatten(
-    chunk2(scanLinesPoints).map(([a, b]: [Vector3, Vector3]) =>
-      traverse(a, b, resolutions, logZoomStep),
-    ),
-  );
+
+  const planeIds: Array<OrthoViewWithoutTD> =
+    viewMode === "orthogonal" ? ["PLANE_XY", "PLANE_XZ", "PLANE_YZ"] : ["PLANE_XY"];
+  let traversedBuckets: Vector3[] = [];
+  for (const planeId of planeIds) {
+    let extent: Vector2;
+    let enlargedHalfExtent: Vector2;
+    let queryMatrix = [...matrix] as Matrix4x4;
+
+    if (viewMode === "orthogonal") {
+      extent = [rects[planeId].width, rects[planeId].height];
+      enlargedHalfExtent = [Math.ceil(extent[0] / 2), Math.ceil(extent[1] / 2)] as Vector2;
+      if (planeId === "PLANE_YZ") {
+        M4x4.mul(matrix, YZ_ROTATION, queryMatrix);
+      } else if (planeId === "PLANE_XZ") {
+        M4x4.mul(matrix, XZ_ROTATION, queryMatrix);
+      }
+    } else {
+      // Buckets adjacent to the current viewport are also loaded so that these
+      // buckets are already on the GPU when the user moves a little.
+      extent = [constants.VIEWPORT_WIDTH, constants.VIEWPORT_WIDTH];
+      const enlargementFactor = 1.1;
+      const enlargedExtent = [
+        constants.VIEWPORT_WIDTH * enlargementFactor,
+        constants.VIEWPORT_WIDTH * enlargementFactor,
+      ];
+      enlargedHalfExtent = [enlargedExtent[0] / 2, enlargedExtent[1] / 2] as Vector2;
+    }
+
+    // Cast a vertical "scan line" and check how many buckets are intersected.
+    // That amount N is used as a measure to cast N + 1 (steps) vertical scanlines.
+    const stepRatePoints = M4x4.transformVectorsAffine(queryMatrix, [
+      [-enlargedHalfExtent[0], -enlargedHalfExtent[1], 0],
+      [-enlargedHalfExtent[0], +enlargedHalfExtent[1], 0],
+    ]);
+    const stepRateBuckets = traverse(
+      stepRatePoints[0],
+      stepRatePoints[1],
+      resolutions,
+      logZoomStep,
+    );
+    const steps = stepRateBuckets.length + 1;
+    const stepSize = [extent[0] / steps, extent[1] / steps];
+    // This array holds the start and end points
+    // of horizontal lines which cover the entire rendered plane.
+    // These "scan lines" are traversed to find out which buckets need to be
+    // sent to the GPU.
+    const zDiff = 10;
```

Comment:
The coordinates are essentially in "screen space" and multiplied with the `queryMatrix`. So, a static value makes somewhat sense in my opinion. However, I'm sure this could be improved somehow. However, since I'm not sure how, I'd leave it as is for the moment :smile: